pages/instructorScanPaper/instructorScanPaper.js

```diff
@@ -0,0 +1,80 @@
+const express = require('express');
+const router = express.Router();
+const path = require('path');
+const debug = require('debug')('prairielearn:' + path.basename(__filename, '.js'));
+
+const { processScrapPaperPdf } = require('../../lib/scrapPaperReader');
+const serverJobs = require('../../lib/server-jobs');
+
+const error = require('../../prairielib/lib/error');
+const ERR = require('async-stacktrace');
+
+router.get('/', (req, res) => {
+  res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+});
+
+router.post('/', function (req, res, next) {
+  if (!res.locals.authz_data.has_course_instance_permission_edit) return next();
+  if (req.body.__action === 'scan_scrap_paper') {
+    if (!req.file) {
+      return next(error.make(400, 'Missing barcoded pdf collection file data'));
+    }
+    if (!res.locals || !res.locals.authn_user || !res.locals.authn_user.user_id) {
+      return next(error.make(400, 'Authn_user required on file-store API'));
+    }
+
+    const options = {
+      course_id: res.locals.course ? res.locals.course.id : null,
+      type: 'decoding_pdf_barcoded_collection',
+      description: 'Load data from local disk',
+    };
+    let jobSequenceId = null;
+
+    serverJobs
+      .createJobSequenceAsync(options)
+      .then((jsId) => {
+        jobSequenceId = jsId;
+
+        const jobOptions = {
+          course_id: res.locals.course ? res.locals.course.id : null,
+          type: 'decode_pdf_collection',
+          description:
+            'Decodes each barcode on each page in the pdf collection used as scrap paper by students.',
+          job_sequence_id: jobSequenceId,
+          last_in_sequence: false,
+        };
+
+        return serverJobs.createJobAsync(jobOptions);
+      })
+      .then((job) => {
```

Comment:
Let's rewrite this with async/await to avoid the need for explicit promise chaining.

Reply:
Does async await work on the route functions? I thought promise chains were needed.

Reply:
Express doesn't support that natively, but we wrap routes with [`express-async-handler`](https://www.npmjs.com/package/express-async-handler) - see [this example](https://github.com/PrairieLearn/PrairieLearn/blob/2f3b359987bebf00d45a8c32d58e3b302d117c4a/pages/administratorQueries/administratorQueries.js#L14).